packages/react/runtime/src/list.ts

```diff
@@ -263,13 +265,58 @@ export function componentAtIndexFactory(
       throw new Error('componentAtIndex called on removed list');
     }
 
-    const childCtx = ctx[cellIndex];
-    if (!childCtx) {
-      throw new Error('childCtx not found');
-    }
-
     const platformInfo = childCtx.__listItemPlatformInfo || {};
 
+    // The lifecycle of this `__extraProps.isReady`:
+    //   0 -> Promise<number> -> 1
+    // 0: The initial state, the list-item is not ready yet, we will send a event to background
+    //    when `componentAtIndex` is called on it
+    // Promise<number>: A promise that will be resolved when the list-item is ready
+    // 1: The list-item is ready, we can use it to render the list
+    if (childCtx.__extraProps?.['isReady'] === 0) {
+      __OnLifecycleEvent([LifecycleConstant.publishEvent, {
+        handlerName: `${childCtx.__id}:__extraProps:onComponentAtIndex`,
+        data: {
+          listID,
+          childCtxId: childCtx.__id,
+        },
+      }]);
+
+      let p: Promise<number>;
+      return (p = new Promise<number>((resolve) => {
```

Comment:
Is is correct to return a `Promise` in `componentAtIndex`? It seems like the Lynx Engine would cast it to a number:

https://github.com/lynx-family/lynx/blob/e795287f127b7ccd5719fb4d17186d5f80c63751/core/renderer/dom/fiber/list_element.cc#L167-L175